webui/react/src/pages/JobQueue/JobQueue.tsx

```diff
@@ -131,7 +131,7 @@ const JobQueue: React.FC<Props> = ({ selectedRp, jobState }) => {
   useEffect(() => {
     fetchAll();
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [settings.tableOffset, settings.tableLimit]);
```

Comment:
it works but it looks a bit hacky. how about this?

```diff
diff --git a/webui/react/src/pages/JobQueue/JobQueue.tsx b/webui/react/src/pages/JobQueue/JobQueue.tsx
index c98a7589b..3b407b4e6 100644
--- a/webui/react/src/pages/JobQueue/JobQueue.tsx
+++ b/webui/react/src/pages/JobQueue/JobQueue.tsx
@@ -14,6 +14,7 @@ import {
   userRenderer,
 } from 'components/Table/Table';
 import { V1SchedulerTypeToLabel } from 'constants/states';
+import usePolling from 'hooks/usePolling';
 import { useSettings } from 'hooks/useSettings';
 import { columns as defaultColumns, SCHEDULING_VAL_KEY } from 'pages/JobQueue/JobQueue.table';
 import { paths } from 'routes/utils';
@@ -128,10 +129,7 @@ const JobQueue: React.FC<Props> = ({ selectedRp, jobState }) => {
     }
   }, [canceler.signal, selectedRp.name, settings, jobState, topJob, updateSettings]);
 
-  useEffect(() => {
-    fetchAll();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [settings.tableOffset, settings.tableLimit]);
+  usePolling(fetchAll, { rerunOnNewFn: true });
 
   const rpTotalJobCount = useCallback(
     (rpName: string) => {
```

Reply:
agreed, this is closer to how we get info elsewhere

Reply:
updated to this